### DISCUSSION TITLE
Add topic title to buffer name

### DISCUSSION THREAD

**Discussion by nbarrientos** [magit/forge#542] [Ideas]
Hi,

Would it make sense to add the topic's title to the buffer name when `forge-topic-setup-buffer` is called?

Something like:

```diff
@@ -407,10 +407,11 @@ defun forge-topic-setup-buffer
   (let* ((repo  (forge-get-repository topic))
          (ident (concat (forge--topic-type-prefix topic)
                         (number-to-string (oref topic number))))
-         (name  (format "*forge: %s/%s %s*"
+         (name  (format "*forge: %s/%s %s \"%s\"*"
                         (oref repo owner)
                         (oref repo name)
-                        ident))
+                        ident
+                        (oref topic title)))
          (magit-generate-buffer-name-function (lambda (_mode _value) name))
          (current-repo (forge-get-repository nil))
          (default-directory (if (and current-repo
```
Currently the buffer name is composed by the topic number and the repository owner/name. This would allow searching for issues or pull requests in the buffer list by title. Personally I find it easier to be able to use (key)words that I remember from the title when unburying buffers ;)

Not sure though that all the topic types that `forge-topic-setup-buffer` has to eventually deal with have the slot `title`.

What do you think?

**Comment by tarsius** [comment]
The title is now included in the header. (Though arguably the format should be customizable, but that isn't what was requested here.)

**Comment by nbarrientos** [comment]
Thank you!
